packages/ethereum-contracts/test/foundry/FoundrySuperfluidTester.sol

```diff
@@ -65,13 +128,35 @@ contract FoundrySuperfluidTester is Test {
         sfDeployer.deployTestFramework();
         sf = sfDeployer.getFramework();
 
-        resolver = sf.resolver;
-
         require(nTesters <= TEST_ACCOUNTS.length, "too many testers");
         N_TESTERS = nTesters;
+
+        // Set the token type being tested
+        string memory tokenType = vm.envOr(TOKEN_TYPE_ENV_KEY, DEFAULT_TEST_TOKEN_TYPE);
+        bytes32 hashedTokenType = keccak256(abi.encode(tokenType));
+
+        testSuperTokenType = hashedTokenType == keccak256(abi.encode("WRAPPER_SUPER_TOKEN"))
```

Comment:
hurts my eyes :) could have just used a 'if else train' too

Reply:
unfortunately a necessity due to: "cannot initialize immutable in if block"